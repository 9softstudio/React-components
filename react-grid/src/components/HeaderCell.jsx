import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class HeaderCell extends Component {
    constructor(props) {
        super(props);

    }

    static propTypes = {
        sortOrder: PropTypes.string,
        sortBy: PropTypes.string,
        onSort: PropTypes.func
    }
 
    static defaultProps = {
        sortOrder: null,
        sortBy: null,
        onSort: null
    }

    onClick = () => {
        if(!this.props.sortBy){
            return;
        }
        const { onSort } = this.props;
        const sortOrder = this.props.sortOrder;

        const newSortOrder = !sortOrder || sortOrder === 'desc' ? 'asc' : 'desc';
        onSort && onSort(this.props.sortBy, newSortOrder);
    }

    render() {
        const { colWidth, sortBy, sortOrder, onSort, children, ...rest } = this.props;
    
        let sortableHeader = null;
        if(sortBy) {
            const upArrow = <div style={{display:'block', color: `${sortOrder === 'asc' ? '#111':'#777'}`}} className={'asc'}></div>
            const downArrow = <div style={{display:'block', marginTop:'2px', color: `${sortOrder === 'desc' ? '#111':'#777'}`}} className={'desc'}></div>
            sortableHeader = (<th {...rest} className={'sortable sorting'} style={{position:'relative', backgroundClip: 'padding-box'}} onClick={this.onClick}>                   
                <div style={{width:'90%'}}>{children}</div>
                <span style={{float: 'right', position:'absolute', top:'40%', right:'2px'}}>
                    {upArrow}
                    {downArrow}
                </span>
                
            </th>)
        }

        return sortableHeader || 
            (<th {...rest}>{children}</th>)
    }
}