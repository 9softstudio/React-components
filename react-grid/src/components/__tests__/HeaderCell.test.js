import React from 'react'
import HeaderCell from '../HeaderCell'
import ShallowRenderer from 'react-test-renderer/shallow'

test('sortable HeaderCell render correctly', () => {
    const renderer = new ShallowRenderer();

    const cell = renderer.render(
        <HeaderCell sortBy='Cell1' sortOrder='asc' colWidth='200' anotherProp='test'>Cell 1</HeaderCell>
    );

    expect(cell).toMatchSnapshot();
})

test('unsortable HeaderCell render correctly', () => {
    const renderer = new ShallowRenderer();

    const cell = renderer.render(
        <HeaderCell colWidth='200' anotherProp='test'>Cell 1</HeaderCell>
    );

    expect(cell).toMatchSnapshot();
})