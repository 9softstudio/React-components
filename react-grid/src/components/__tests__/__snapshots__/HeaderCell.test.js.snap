// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortable HeaderCell render correctly 1`] = `
<th
  anotherProp="test"
  className="sortable sorting"
  onClick={[Function]}
  style={
    Object {
      "backgroundClip": "padding-box",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "width": "90%",
      }
    }
  >
    Cell 1
  </div>
  <span
    style={
      Object {
        "float": "right",
        "position": "absolute",
        "right": "2px",
        "top": "40%",
      }
    }
  >
    <div
      className="asc"
      style={
        Object {
          "color": "#111",
          "display": "block",
        }
      }
    />
    <div
      className="desc"
      style={
        Object {
          "color": "#777",
          "display": "block",
          "marginTop": "2px",
        }
      }
    />
  </span>
</th>
`;

exports[`unsortable HeaderCell render correctly 1`] = `
<th
  anotherProp="test"
>
  Cell 1
</th>
`;
